state-chain/chains/src/dot.rs

```diff
@@ -314,6 +314,38 @@ impl FeeEstimationApi<Polkadot> for PolkadotTrackedData {
 	}
 }
 
+#[derive(
+	Encode, Decode, TypeInfo, Clone, RuntimeDebug, Default, PartialEq, Eq, Serialize, Deserialize,
+)]
+pub struct PolkadotTransactionMetadata {
+	pub tx_ref: PolkadotTransactionSignature,
+}
+
+impl<C: Chain<TransactionHashItem = PolkadotTransactionSignature>> TransactionMetadataHash<C>
+	for PolkadotTransactionMetadata
+{
+	fn get_transaction_hash(&self) -> <C as Chain>::TransactionHashItem {
+		self.tx_ref.clone()
+	}
+}
+impl<C: Chain> TransactionMetadata<C> for PolkadotTransactionMetadata {
+	fn extract_metadata(_transaction: &<C as Chain>::Transaction) -> Self {
+		Default::default()
+	}
+
+	fn verify_metadata(&self, _expected_metadata: &Self) -> bool {
+		true
+	}
+}
+
+#[derive(
+	Encode, Decode, TypeInfo, Clone, RuntimeDebug, Default, PartialEq, Eq, Serialize, Deserialize,
+)]
+pub struct PolkadotTransactionSignature {
+	pub block_hash: H256,
```

Comment:
we should use the `PolkadotBlockNumber` here instead. As it's only 32 bits, compared to the 32 byte hash - but still uniquely identifies the transaction.